src/main/java/com/spotify/docker/BuildMojo.java

```diff
@@ -493,6 +533,36 @@ private void createDockerFile(final String directory, final List<String> filesTo
     if (maintainer != null) {
       commands.add("MAINTAINER " + maintainer);
     }
+
+    if (env != null) {
+      final List<String> sortedKeys = Ordering.natural().sortedCopy(env.keySet());
+      for (String key : sortedKeys) {
+        final String value = env.get(key);
+        commands.add(String.format("ENV %s %s", key, value));
+      }
+    }
+
+    if (workdir != null) {
+      commands.add("WORKDIR " + workdir);
+    }
+
+    for (String file : filesToAdd) {
+      commands.add(String.format("ADD %s %s", file, file));
+    }
+
+    if (runList != null && !runList.isEmpty()) {
+      commands.add("RUN " + Joiner.on(" &&\\\n\t").join(runList));
```

Comment:
Hm not sure. [Docker best practices guide][1] implies things like apt-get update and apt-get install should be in one line for cache busting purposes. As a user, I probably would guess they'd be separate RUN commands.

@dawidmalina What was the rationale behind one RUN command and hence one layer?

  [1]: https://docs.docker.com/articles/dockerfile_best-practices/#run-https-docs-docker-com-reference-builder-run